### PR TITLE
Fix for permissions not reflecting when adding duplicate

### PR DESCRIPTION
**Pull request by sarathmohan4**

### CODE REVIEW

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SecretService.java [hunk 1]
```diff
@@ -101,7 +104,16 @@
 			return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("{\"errors\":[\"Invalid request.Check json data\"]}");
 		}
 		if(ControllerUtil.isPathValid(path)){
-			//if(ControllerUtil.isValidSafe(path,token)){
+			if (!hasExplicitWritePermission(token, userDetails.getUsername(), secret.getPath())) {
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SecretService.java]: You may need to add comment why we are doing this

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SecretService.java [hunk 2]
```diff
@@ -136,6 +145,38 @@
...
+	/**
+	 * To check if user has explicit write permission
+	 * @param token
+	 * @param userName
+	 * @param path
+	 * @return
+	 */
+	private boolean hasExplicitWritePermission(String token, String userName, String path) {
+		Response userResponse;
+		if (TVaultConstants.USERPASS.equals(vaultAuthMethod)) {
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SecretService.java]: I remember we added a factory based implementation for authentication/branching. Is it not applicable here?

File: tvaultapi/src/test/java/com/tmobile/cso/vault/api/service/SecretServiceTest.java [hunk 3]
```diff
@@ -99,12 +114,65 @@ public void test_write_successfully() {
...
+
+        Response userResponse = getMockResponse(HttpStatus.OK, true, "{\"data\":{\"bound_cidrs\":[],\"max_ttl\":0,\"policies\":[\"default\",\"w_shared_mysafe01\",\"w_shared_mysafe01\"],\"ttl\":0,\"groups\":\"admin\"}}");
+        List<String> resList = new ArrayList<>();
+        resList.add("default");
+        resList.add("w_shared_mysafe01");
+        resList.add("w_shared_mysafe02");
+        try {
+            when(ControllerUtil.getPoliciesAsListFromJson(any(), any())).thenReturn(resList);
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (sivanag1974) [inline tvaultapi/src/test/java/com/tmobile/cso/vault/api/service/SecretServiceTest.java]: what is the use of printStackTrace() and why is it required here?

Comment (sarathmohan4) [reply tvaultapi/src/test/java/com/tmobile/cso/vault/api/service/SecretServiceTest.java]: Removed

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SecretService.java (hunk 1 above)
Comment (sarathmohan4) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SecretService.java]: Updated comment

**Pull request merged**